src/components/common/motd-banner/motd-banner.tsx

```diff
@@ -1,41 +1,56 @@
 /*
- SPDX-FileCopyrightText: 2021 The HedgeDoc developers (see AUTHORS file)
-
- SPDX-License-Identifier: AGPL-3.0-only
+ * SPDX-FileCopyrightText: 2021 The HedgeDoc developers (see AUTHORS file)
+ *
+ * SPDX-License-Identifier: AGPL-3.0-only
  */
 
 import equal from 'fast-deep-equal'
-import React from 'react'
+import React, { useCallback } from 'react'
 import { Alert, Button } from 'react-bootstrap'
 import { useSelector } from 'react-redux'
 import { Link } from 'react-router-dom'
 import { ApplicationState } from '../../../redux'
 import { setBanner } from '../../../redux/banner/methods'
 import { ForkAwesomeIcon } from '../fork-awesome/fork-awesome-icon'
-import { ShowIf } from '../show-if/show-if'
+import { BANNER_LOCAL_STORAGE_KEY } from '../../application-loader/initializers/fetch-and-set-banner'
 
 export const MotdBanner: React.FC = () => {
   const bannerState = useSelector((state: ApplicationState) => state.banner, equal)
 
-  const dismissBanner = () => {
-    setBanner({ ...bannerState, show: false })
-    window.localStorage.setItem('bannerTimeStamp', bannerState.timestamp)
+  const dismissBanner = useCallback(() => {
+    if (bannerState.banner?.etag) {
+      window.localStorage.setItem(BANNER_LOCAL_STORAGE_KEY, bannerState.banner.etag)
+    }
+    setBanner({
+      banner: {
+        text: undefined,
+        etag: null
+      }
+    })
+  }, [bannerState])
+
+  if (!bannerState.banner) {
+    return null
+  }
+
+  if (!bannerState.banner.text) {
+    return <span data-cy={ 'no-motd-banner' }/>
   }
 
   return (
-    <ShowIf condition={ bannerState.show }>
-      <Alert variant='primary' dir='auto' className='mb-0 text-center d-flex flex-row justify-content-center'>
-        <Link to='/s/banner' className='flex-grow-1 align-self-center text-black'>
-          { bannerState.text }
-        </Link>
-        <Button
-          variant='outline-primary'
-          size='sm'
-          className='mx-2'
-          onClick={ dismissBanner }>
-          <ForkAwesomeIcon icon='times'/>
-        </Button>
-      </Alert>
-    </ShowIf>
+    <Alert data-cy={ 'motd-banner' } variant="primary" dir="auto"
+           className="mb-0 text-center d-flex flex-row justify-content-center">
+      <Link to="/s/banner" className="flex-grow-1 align-self-center text-black">
+        { bannerState.banner.text }
+      </Link>
```

Comment:
The link to /s/banner doesn't make sense anymore when the banner content isn't the same thing.
Maybe we want to allow placing a custom URL in the last line of the banner.txt and use that as link?

Reply:
Hm. We could use markdown for the banner and could define this in it's frontmatter.

Reply:
I removed the Link for now. If we still want to have it then we can readd that in another PR.